src/ja/server/scheduler/algorithm.py

```diff
@@ -0,0 +1,84 @@
+from abc import ABC, abstractmethod
+from typing import List, Optional, Tuple
+from ja.common.job import Job
+from ja.server.database.database import ServerDatabase
+from ja.server.database.types.work_machine import WorkMachine
+
+
+class CostFunction(ABC):
+    """
+    A CostFunction is a function which determines the order in which jobs are scheduled.
+    """
+
+    @abstractmethod
+    def calculate_cost(self, job: Job) -> float:
+        """!
+        Calculate the cost of a job.
+
+        The cost of a job is an integer. Higher values mean higher priority for
```

Comment:
Docstring still says integer.
Also, going by the convention for cost functions, the Scheduler would be supposed to pick the job with the lowest cost function value, not the one with the highest.
(We can just use negative effective priority for this.)

Reply:
Fixed.